### PR TITLE
React Templating

### PR DESCRIPTION
**Pull request by jimniels**
Move from EJS to React for templating.

### CODE REVIEW

File: TestComponent.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
+
+export default function Component() {
+  return <div>hi</div>;
+}
```
Comment (damassi) [inline TestComponent.js]: `TestComponent.tsx` 😉

Comment (damassi) [reply TestComponent.js]: 👋

Comment (jimniels) [reply TestComponent.js]: lol

hey, get out of here you! this PR is very much WIP and is for my 👀 only haha

Comment (damassi) [reply TestComponent.js]: followed the trail from your blog. I still need to try out es module support...

hope yr well jim :) :)

Comment (jimniels) [reply TestComponent.js]: At first I tried to do this straight up through node, just `node --experimental-modules metalsmith.mjs`. The only reason I was using babel was to transpire my JSX. 

Boy oh boy, that didn't work though. I had to throw more babel at it and finally [esm](https://github.com/standard-things/esm#readme) to get it working how I expected. I'll be updating this PR pretty soon. That should get all the plumbing in place. Then I just need to rewrite all my .ejs files to be .jsx.

**Pull request merged**